### ISSUE TITLE
Crash after updating vscode extension to 1.57.0 

### ISSUE THREAD

**Issue by ahbali** [spring-projects/sts4#1355]
**Describe the bug**
Visual Studio Code 1.93.1.
OS: Ubuntu 22.04.
the spring tools extension 1.56.0 was working fine, but after updating to 1.57.0 the extension started crashing when i open one of my projects.

PS: my other projects are working fine with the updated version.

[spring_tools_bug_new.log](https://github.com/user-attachments/files/17086056/spring_tools_bug_new.log)

what is strange is that the language server logs appeared in my project's Logback RollingFileAppender

[sdmbe.log](https://github.com/user-attachments/files/17086216/sdmbe.log)

I would have shared the project code if it was not an enterprise project, sorry.

**Comment by BoykoAlex** [comment]
I don't know what to make of it yet... Your logging setup seems suspicious... 

Statements in the log such as:
```
22:20:08,470 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
22:20:08,472 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/home/username/.vscode/extensions/vmware.vscode-spring-boot-1.57.0/language-server/lib/commons-language-server-1.57.0-SNAPSHOT.jar!/logback.xml]
22:20:08,476 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@77cf3f8b - URL [jar:file:/home/username/.vscode/extensions/vmware.vscode-spring-boot-1.57.0/language-server/lib/commons-language-server-1.57.0-SNAPSHOT.jar!/logback.xml] is not of type file
22:20:08,580 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - Processing appender named [STDOUT]
22:20:08,581 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:20:08,589 |-INFO in ch.qos.logback.core.model.processor.ImplicitModelHandler - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
22:20:08,614 |-INFO in ch.qos.logback.classic.model.processor.RootLoggerModelHandler - Setting level of ROOT logger to INFO
22:20:08,614 |-INFO in ch.qos.logback.core.model.processor.AppenderRefModelHandler - Attaching appender named [STDOUT] to Logger[ROOT]
22:20:08,614 |-INFO in ch.qos.logback.core.model.processor.DefaultProcessor@1df98368 - End of configuration.
```

Don't think I'm seeing these on my machine...

**Comment by ahbali** [comment]
@BoykoAlex 
i might have found how to reproduce the bug:
having an application.yml in the root of the project, where logging.config is set to logback-spring.xml file.
I created a repo with a very simple spring-boot project that reproduces the crash of the extension for me: https://github.com/ahbali/spring-tools-bug

**Comment by BoykoAlex** [comment]
Looks like it is the switch to the new LS JAR layout that surfaced this issue. Seems the new layout would look for config files in the CWD of the process.
The CWD of the LS process is the the workspace root for VSCode and user home directory for Eclipse it is the `user.dir` property from the running eclipse process, i.e. folder hosting eclipse executable. 
This explains the reason for `application.yml` located in the root folder of the workspace root to affect the LS process logging settings.
Looks as we need to take better care of the CWD for the LS process from now and on.

**Comment by BoykoAlex** [comment]
Fixed with c2454a86deaf840ccb4bb83b67b1d9e3ff7775f1
